File: Solved/Soobasa Va Dostan/SoobasaVaDostan.java
import java.util.Scanner;

/*
Question title = subasa va doostan - سوباسا و دوستان
https://quera.ir/problemset/108669/
 */

public class SoobasaVaDostan {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        String[] input = sc.nextLine().split("\\s");
        int countGoal = Integer.parseInt(input[0]);
        int firstHalf = Integer.parseInt(input[1]);
        int secondHalf = Integer.parseInt(input[2]);

        String[] goals = sc.nextLine().split("\\s");

        int sum = calculate(goals, firstHalf, secondHalf);

        if (sum < countGoal)
            System.out.println("NO");
        else
            System.out.println("YES");
    }

    // returns sum of goals in first half and second half
    private static int calculate(String[] goals, int firstHalf, int secondHalf) {
        int firstTimeGoals = 0;
        int secondTimeGoals = 0;

        boolean done = false;
        int previousTimeInFirstHalf = Integer.parseInt(goals[0]);
        int previousTimeInSecondHalf = 45;

        for (String i : goals) {
            int goal = Integer.parseInt(i);

            // if the goal is in the first half
            if (goal <= 45 + firstHalf && goal >= previousTimeInFirstHalf && !done) {
                firstTimeGoals++;
            }
            // if the goal is in the second half
            else if (goal >= 45 && goal <= 90 + secondHalf && goal >= previousTimeInSecondHalf) {
                secondTimeGoals++;
                done = true;
                previousTimeInSecondHalf = goal;
            }
            previousTimeInFirstHalf = goal;
        }
        return firstTimeGoals + secondTimeGoals;
    }
}

File: Solved/Soobasa Va Dostan/Soobasa Va Dostan.py
# سوباسا و دوستان
# ID : 108669
# https://quera.org/problemset/108669/
# https://b2n.ir/u73922

def calculate(goals, firstHalf, secondHalf):
    firstTimeGoals = 0
    secondTimeGoals = 0

    done = False
    previous_time_in_first_half = int(goals[0])
    previous_time_in_second_half = 45

    for i in goals:
        goal = int(i)

        if 45 + firstHalf >= goal >= previous_time_in_first_half and not done:
            firstTimeGoals += 1

        elif 45 <= goal <= 90 + secondHalf and goal >= previous_time_in_second_half:
            secondTimeGoals += 1
            done = True
            previous_time_in_second_half = goal

        previous_time_in_first_half = goal

    return firstTimeGoals + secondTimeGoals


inputer = [int(i) for i in input().split(" ")]
countGoal, firstHalf, secondHalf = inputer
goals = input().split(" ")

sum = calculate(goals, firstHalf, secondHalf)

if sum < countGoal:
    print("NO")

else:
    print("YES")
